lib/src/main/java/com/otaliastudios/transcoder/internal/thumbnails/DefaultThumbnailsEngine.kt

```diff
@@ -77,6 +77,8 @@ class DefaultThumbnailsEngine(
     }
 
     private val stubs = ArrayDeque<Stub>()
+    private val bucketListMap = LinkedHashMap<Long, ArrayList<Stub>>()
```

Comment:
We can create the final list in runtime in the reorder method itself

Reply:
We should maintain as minimal state as possible.
And in this case, we don't get a huge advantage by maintaining this

Reply:
initially i had this inside only. but it didn't make sense creating new list for every request @sahilbajaj . that's why moved it to global

Reply:
this will happen for every batch.
And lists will be small in size.
It's not a considerable optimization for sure. Not worth it